src/ILVerification/src/InterfaceVerifier.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Resources;
+using Internal.TypeSystem;
+using Internal.TypeSystem.Ecma;
+
+namespace ILVerify
+{
+    internal class InterfaceVerifier
```

Comment:
Interface verification seems to be too narrow concept. I would look at this as a more general type verification. There is likely to be a lot of overlap between what needs to verified for interfaces and for the other types.